packages/fdc3-agent-proxy/src/channels/DefaultChannelSupport.ts

```diff
@@ -93,8 +113,14 @@ export class DefaultChannelSupport implements ChannelSupport {
       },
     };
     const response = await this.messaging.exchange<GetOrCreateChannelResponse>(request, 'getOrCreateChannelResponse');
-
-    return new DefaultChannel(this.messaging, id, 'app', response.payload.channel?.displayMetadata!!);
+    //handle successful responses - errors will already have been thrown by exchange above
+    if (response.payload.channel) {
+      const out = new DefaultChannel(this.messaging, id, 'app', response.payload.channel.displayMetadata);
+      return out;
+    } else {
+      console.error('Invalid response from Desktop Agent to getUserChannelsRequest!', response);
```

Comment:
Doesn't `console.error` also throw an error? Wouldn't that mean that the next line is both redundant and unreachable?

Reply:
nope, it just prints to stderr so you need to log before throwing (as a log message after throwing is unreachable).